src/components/Calculator.js

```diff
@@ -1,33 +1,66 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 import './calculator.css';
 
 // eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.dispspalyValue = this.dispspalyValue.bind(this);
+  }
+
+  dispspalyValue(e) {
+    const btnName = e.target.innerText;
+    const obj = calculate(this.state, btnName);
+    this.setState(() => ({
+      total: obj.total,
+      next: obj.next,
+      operation: obj.operation,
+    }));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
+    const op = operation === '%' ? '%' : operation;
```

Comment:
- [ ] The modulo (%) operation is not working as intended. Trying to operate `9 % 2` is returning:
 _Can't find modulo as can't divide by 0_ instead of 1. Kindly ensure that the % operator is working correctly.

**Modulo operation error:**
![image](https://user-images.githubusercontent.com/12745474/189238213-e951c2a8-7bdf-44b2-9d2f-159caf2ecd58.png)